src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/order/UniqueOrderList.java

```diff
@@ -74,6 +81,50 @@ public void setOrders(List<Order> orders) {
         internalList.setAll(orders);
     }
 
+    /**
+     * Returns the top ten clients with the most orders.
+     *
+     * @return Top ten clients with the most orders.
+     */
+    public List<Map.Entry<Pair<GenericString, Phone>, Integer>> getTopTenClients() {
+        HashMap<Pair<GenericString, Phone>, Integer> clientOrderCountKeyValuePair =
+                getClientOrderCountKeyValuePair();
+
+        List<Map.Entry<Pair<GenericString, Phone>, Integer>> topTenClients =
+                getTopTenClientsFromHashMap(clientOrderCountKeyValuePair);
+
+        return topTenClients;
+    }
+
+    private HashMap<Pair<GenericString, Phone>, Integer> getClientOrderCountKeyValuePair() {
+        HashMap<Pair<GenericString, Phone>, Integer> clientOrderCountKeyValuePair = new HashMap<>();
+
+        internalList.stream().forEach(order -> {
+            GenericString clientName = order.getClientName();
+            Phone clientPhone = order.getClientPhone();
+            Pair<GenericString, Phone> clientNameAndPhone = new Pair<>(clientName, clientPhone);
+            if (clientOrderCountKeyValuePair.containsKey(clientNameAndPhone)) {
+                int orderCount = clientOrderCountKeyValuePair.get(clientNameAndPhone);
+                clientOrderCountKeyValuePair.put(clientNameAndPhone, ++orderCount);
+            } else {
+                clientOrderCountKeyValuePair.put(clientNameAndPhone, 1);
+            }
+        });
```

Comment:
Can we change this whole thing?
```suggestion
        Map<Pair<GenericString, Phone>, Long> clientToOrderCountMap = internalList.stream()
                .collect(Collectors.groupingBy((order) ->
                        new Pair<>(order.getClientName(), order.getClientPhone()), Collectors.counting()));
```